Mathlib/Data/Fintype/BigOperators.lean

```diff
@@ -198,7 +198,8 @@ theorem Finset.prod_univ_sum [DecidableEq α] [Fintype α] [CommSemiring β] {δ
 gives `(a + b)^n`. The "good" proof involves expanding along all coordinates using the fact that
 `x^n` is multilinear, but multilinear maps are only available now over rings, so we give instead
 a proof reducing to the usual binomial theorem to have a result over semirings. -/
-theorem Fintype.sum_pow_mul_eq_add_pow (α : Type _) [Fintype α] {R : Type _} [CommSemiring R]
+theorem Fintype.sum_pow_mul_eq_add_pow (α : Type _) [DecidableEq α] [Fintype α] {R : Type _}
+  [CommSemiring R]
```

Comment:
If we don't need `[DecidableEq]` to **state** a theorem, only to **prove** it, then we use classical reasoning.